components/inventory.ts

```diff
@@ -384,13 +385,72 @@ function filterAllowedContent(gameVersion: GameVersion, entP: string[]) {
     }
 }
 
+/**
+ * Given an inventory and a sublocation, returns a new inventory with
+ * the default suit for the sublocation added if it is not already present.
+ * If the sublocation is undefined, the inputted inventory is returned.
+ * Otherwise, a new inventory is cloned, appended with the default suit, and returned.
+ * Either way, the inputted inventory is not modified.
+ * @param profileId The profileId of the player
+ * @param gameVersion  The game version
+ * @param inv  The inventory to update
+ * @param sublocation  The sublocation to check for a default suit
+ * @returns  The updated inventory
+ */
+function updateWithDefaultSuit(
+    profileId: string,
+    gameVersion: GameVersion,
+    inv: InventoryItem[],
+    sublocation: Unlockable,
+): InventoryItem[] {
+    if (sublocation === undefined) {
+        return inv
+    }
+
+    // We need to add a suit, so need to copy the cache to prevent modifying it.
+    const newInv = [...inv]
+
+    // Yes this is slow. We should organize the unlockables into a { [Id: string]: Unlockable } map.
+    const locationSuit = getVersionedConfig<Unlockable[]>(
+        "allunlockables",
+        gameVersion,
+        true,
+    ).find((u) => u.Id === getDefaultSuitFor(sublocation))
```

Comment:
@RDIL thoughts on doing this?

Reply:
Unless we refactor the whole codebase into using a map for this, the time spent generating a map for this will probably be the same tbh

Reply:
Yeah, I just edited my original message adding that we'd need to change anything that uses it to use `Object.values`. We could also just change the static file to a map.

Reply:
I've actually just had a look at everywhere that uses it, and it would probably reduce the number of filter/find calls we have to do, **but** if an SMF unlockables is used, we'd have to generate a map for it.